infra/gravity/node_commands.go

```diff
@@ -378,6 +380,25 @@ func (g *gravity) SetInstaller(ctx context.Context, installerUrl string, subdir
 	return nil
 }
 
+// ExecScript will transfer and execute script provided with given args
+func (g *gravity) ExecScript(ctx context.Context, scriptUrl string, args []string) error {
+	log := g.Logger().WithFields(logrus.Fields{
+		"script": scriptUrl, "args": args})
+
+	log.Debug("execute")
+
+	spath, err := sshutils.TransferFile(ctx, g.Client(), log,
+		scriptUrl, defaults.TmpDir, g.param.env)
+	if err != nil {
+		log.WithError(err).Error("failed to transfer script")
+		return trace.Wrap(err)
+	}
+
+	err = sshutils.Run(ctx, g.Client(), log,
+		fmt.Sprintf("sudo /bin/bash -x %s %s", spath, strings.Join(args, " ")), nil)
```

Comment:
For why use debug(-x) here? And why bash, not sh... maybe better to copy script, chmod +x on the node end execute. Inside script we can use preferred shell via shebang.

Reply:
cause otherwise if your script would fail, you'll have no clue why. I've seen it many times with bootstrap scripts which silently quit and you have no idea which program caused that cryptic error message.

`bash` because it's available on all VMs we use, and it has some richer syntax.